File: lib/home-card-tones.ts
/** Subtle premium tints for home marketplace cards. Length is odd so
 *  neighbors differ in common 2/3/4/6-column grids. */
export const HOME_CARD_TONES = [
  '#F3E7C8', // champagne
  '#D8F3E6', // mint
  '#F8E3D4', // peach
  '#E7F1D8', // citrus leaf
  '#F6EDDF', // warm sand
] as const;

export type HomeCardTone = (typeof HOME_CARD_TONES)[number];

export function homeCardTone(index: number): HomeCardTone {
  const len = HOME_CARD_TONES.length;
  return HOME_CARD_TONES[((index % len) + len) % len];
}

File: Mobile App/lib/theme/home_card_tones.dart
import 'package:flutter/material.dart';

/// Subtle premium tints for shop product cards. Length is odd so
/// neighbors differ in the 2-column grid (same sequence as the web home).
const homeCardTones = <Color>[
  Color(0xFFF3E7C8), // champagne
  Color(0xFFD8F3E6), // mint
  Color(0xFFF8E3D4), // peach
  Color(0xFFE7F1D8), // citrus leaf
  Color(0xFFF6EDDF), // warm sand
];

Color homeCardTone(int index) {
  final len = homeCardTones.length;
  return homeCardTones[((index % len) + len) % len];
}
